assets/javascripts/discourse/components/modal/diff-modal.gjs

```diff
@@ -1,15 +1,76 @@
 import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
 import { action } from "@ember/object";
+import { next } from "@ember/runloop";
+import { inject as service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
+import CookText from "discourse/components/cook-text";
 import DButton from "discourse/components/d-button";
 import DModal from "discourse/components/d-modal";
+import { ajax } from "discourse/lib/ajax";
 import i18n from "discourse-common/helpers/i18n";
 
 export default class ModalDiffModal extends Component {
+  @service currentUser;
+  @tracked loading = false;
+  @tracked diff;
+  suggestion = "";
+
+  PROOFREAD_ID = -303;
+
+  constructor() {
+    super(...arguments);
+    this.diff = this.args.model.diff;
+
+    next(() => {
+      if (this.args.model.selected && this.args.model.toolbarEvent) {
+        this.loadDiff();
+      }
+    });
+  }
+
+  async loadDiff() {
+    this.loading = true;
+
+    const suggestion = await ajax("/discourse-ai/ai-helper/suggest", {
+      method: "POST",
+      data: {
+        mode: this.PROOFREAD_ID,
+        text: this.selectedText,
+        force_default_locale: true,
+      },
+    });
+
+    this.diff = suggestion.diff;
+    this.suggestion = suggestion.suggestions[0];
+    this.loading = false;
+  }
+
+  get selectedText() {
+    const selected = this.args.model.selected;
+    let selection;
+    if (selected.value === "") {
+      selection = selected.pre + selected.post;
+    } else {
+      selection = selected.value;
+    }
+
+    return selection;
+  }
+
   @action
   triggerConfirmChanges() {
     this.args.closeModal();
-    this.args.model.confirm();
+    if (this.args.model.confirm) {
+      this.args.model.confirm();
+    }
```

Comment:
```suggestion
    this.args.model?.confirm();
```

Reply:
I believe the equivalent would be `this.args.model.confirm?.()`

Reply:
Oh yeah you're right, good catch Renato!

Reply:
yeah I don't think this syntax is allowed? even if it was, it feels a bit too weird visually

Reply:
It is allowed, but yes in this case it's too weird, I was thinking of the case where `model` might be null when doing this comment. Here is an example of it working:

```javascript
let x = { y: { z: function () { console.log("alert"); } } }
x.y.z?.()
=> alert
```

Reply:
Yeah it's part of the optional chaining spec and widely supported.

The difference is that `this.args.model?.confirm()` will fail with a present model but absent `confirm` fn, while `this.args.model.confirm?.()` is exactly the same as the original code, checking only for the fn presence

I don't find it too weird (ok, ok, I even like it! 😄), I feel like parsing it is the same as any optional chaining